src/App.tsx

```diff
@@ -17,18 +29,32 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter setTodos={setTodos} />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  setModalVisible={setIsModalVisible}
+                  setSelectedTodo={setTodo}
+                  todos={todos}
+                  selectedTodo={todo}
```

Comment:
Consider using optional chaining when accessing properties of potentially null or undefined objects, e.g., `todos?.map(...)` and `selectedTodo?.id === todo.id && 'has-background-info-light'`.